templates/default/thin.monitrc.erb

```diff
@@ -0,0 +1,7 @@
+<% pid_dir = "/run/lock/#{@app_shortname}/" %>
+
+check process <%= @appserver_name %>_<%= @app_shortname %> with pidfile <%= pid_dir %><%= @appserver_name %>.pid
+start program = "/bin/sh -c 'cd <%= File.join(@deploy_to, 'current') %> && <%= @environment.map {|k,v| "#{k}=\"#{v}\""}.join(' ') %> <%= @appserver_command %> | logger -t <%= @appserver_name %>-<%= @app_shortname %>'" as uid "<%= node['deployer']['user'] %>" and gid "<%= node['deployer']['group'] %>" with timeout 90 seconds
+stop program = "/bin/sh -c 'cat <%= pid_dir %><%= @appserver_name %>.pid | xargs --no-run-if-empty kill -QUIT; sleep 5'" as uid "<%= node['deployer']['user'] %>" and gid "<%= node['deployer']['group'] %>"
+stop restart = "/bin/sh -c 'cat <%= pid_dir %><%= @appserver_name %>.pid | xargs --no-run-if-empty kill -USR2; sleep 5'" as uid "<%= node['deployer']['user'] %>" and gid "<%= node['deployer']['group'] %>"
```

Comment:
TODO: Actually, I'm not sure this is the right way to restart thin. Find the right way or just remove the 'restart' line, in which case monit will fall back to a `stop-start`.

Reply:
@ajgon if you are familiar with Thin, you might have some insight to this one.
https://github.com/macournoyer/thin/blob/0712d603a31d97b9fa8a0260da400da2e4217d60/lib/thin/server.rb#L247 suggests that USR2 is not actually handled.